### PR TITLE
WP 37 - Auth/API Timeouts and better error messages

### PR DESCRIPTION
**Pull request by mmcgahan**
This PR applies auth and API endpoint timeouts that will throw errors when external services don't respond in a reasonable amount of time. It also `catch`es these timeout errors and attempts to gracefully continue request execution. It is up to the consuming application to render errors appropriately.

### CODE REVIEW

File: apiProxy/api-proxy.js [hunk 1]
```diff
@@ -244,14 +245,26 @@ const apiProxy$ = ({ baseUrl, duotoneUrls }) => {
 		// to build the query-specific API request options object
 		const apiConfigToRequestOptions = buildRequestArgs(externalRequestOpts);
 
+		const apiTimeout = 5000;
```
Comment (sadafie) [inline apiProxy/api-proxy.js]: Can we stick this at the top, or somewhere where it's more easily discoverable?

File: plugins/anonAuthPlugin.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Boom from 'boom';
 import chalk from 'chalk';
 import Rx from 'rxjs';
 
+const AUTH_TIMEOUT = 5000;
```
Comment (sadafie) [inline plugins/anonAuthPlugin.js]: Actually can we use a configuration for this value? So it's defined in fewer spots.

File: apiProxy/api-proxy.js [hunk 3]
```diff
@@ -3,7 +3,6 @@ import Rx from 'rxjs';
 const externalRequest$ = Rx.Observable.bindNodeCallback(externalRequest);
 
 import * as apiConfigCreators from './apiConfigCreators';
-import { catchAndReturn$ } from '../util/rxUtils';
```
Comment (sadafie) [inline apiProxy/api-proxy.js]: Method no longer used. Remove from rxUtils?

File: plugins/anonAuthPlugin.js (hunk 2 above)
Comment (mmcgahan) [inline plugins/anonAuthPlugin.js]: yes - I'll put it [here](https://github.com/meetup/meetup-web-platform/blob/master/util/config.js)

**Comment by mmcgahan** [comment]
oooh, gotta set up standalone tests for `fetchUtils.js` to keep up the coverage! I'll do that in a separate branch, but this PR won't be mergeable until that merges and addresses the coverage decline

**Comment by mmcgahan** [comment]
`Prelim` because I'd like #25 to be merged first (its code changes have been pulled into this branch)

**Comment by sadafie** [comment]
Merged 25, this is looking good too

**Pull request merged**